client/domain/TaskObjectFactorySpec.js

```diff
@@ -0,0 +1,65 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for TaskObject domain objects.
+ */
+
+describe('TaskObjectFactory', function() {
+  var QuestionObjectFactory;
+  var question;
+
+  var TITLE = 'title';
+  var STARTER_CODE = 'starterCode';
+  var AUXILIARY_CODE = 'auxiliaryCode';
+  var OUTPUT_FUNCTION_NAME_1 =
```

Comment:
Actually, the expected value is more like "AuxiliaryCode.splitString" or similar. The idea is that there are System and AuxiliaryCode classes, each of which have some custom methods. The question is providing a reference to this method so that it can be invoked on the test cases during execution.